fili-core/src/main/java/com/yahoo/bard/webservice/web/util/PaginationParameters.java

```diff
@@ -90,6 +93,11 @@ private int parseParameter (String parameter, String parameterName) throws BadPa
             throw new BadPaginationException(errorMessage.format(parameterName));
         }
         try {
+            if (parameter.equals(FIRST)) {
+                return 1;
+            } else if (parameter.equals(LAST)) {
+                return LAST_PAGE;
```

Comment:
Ok I wasn't entirely sure how much the idea of pagination was exposed in the other parts of the system which is why my preference was towards the `Either`. 

I think `-1` will be okay since it's not exposed but it should definitely be internal only and not be allowed in the api.